File: LC1.java
// Solution to LeetCode 1: Two Sum
// Category: Arrays and Hashing
// Difficulty: Easy

import java.util.HashMap;

public class LC1 {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        HashMap<Integer,Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++)
        {
            if (map.containsKey(target-nums[i]))
            {
                ans[0] = i;
                ans[1] = map.get(target-nums[i]);
                return ans;
            }
            map.put(nums[i],i);
        }
        return ans;
    }
}

File: Python/LC1.py
# LeetCode 1: Two Sum
# Difficulty: Easy
# Arrays
# Time Complexity: O(n)

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        elements = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in elements:
                return [elements[complement], i]
            elements[nums[i]] = i
        return []